kie-maven-takari-plugin/src/test/java/org/drools/compiler/xpath/tobeinstrumented/XPathTest.java

```diff
@@ -0,0 +1,660 @@
+        byte[] tmfileBytecode = enhancer.injectReactive("org.drools.compiler.xpath.tobeinstrumented.model.TMFile");
+        byte[] tmfilesetBytecode = enhancer.injectReactive("org.drools.compiler.xpath.tobeinstrumented.model.TMFileSet");
+        byte[] tmdirectoryBytecode = enhancer.injectReactive("org.drools.compiler.xpath.tobeinstrumented.model.TMDirectory");
+        byte[] pojoWithCollectionsBytecode = enhancer.injectReactive("org.drools.compiler.xpath.tobeinstrumented.model.PojoWithCollections");
+
+        ClassPool cp2 = new ClassPool(null);
+        cp2.appendSystemPath();
+        cp2.appendClassPath(BytecodeInjectReactive.classpathFromClass(ReactiveObject.class));
+
+        loadClassAndUtils(cp2,
+                          personBytecode);
+        loadClassAndUtils(cp2,
+                          childBytecode);
+        loadClassAndUtils(cp2,
+                          schoolBytecode);
+        loadClassAndUtils(cp2,
+                          tmfileBytecode);
+        loadClassAndUtils(cp2,
+                          tmfilesetBytecode);
+        loadClassAndUtils(cp2,
+                          tmdirectoryBytecode);
+        loadClassAndUtils(cp2,
+                          pojoWithCollectionsBytecode);
+    }
+
+    private static void loadClassAndUtils(ClassPool cp,
+                                          byte[] bytecode) throws Exception {
+        CtClass theCtClass = cp.makeClass(new ByteArrayInputStream(bytecode));
+        Class<?> class1 = theCtClass.toClass();
+
+        LOG.info("Bytecode-injected class for {} now having the following methods:",
+                 theCtClass.getName());
+        for (Method m : class1.getMethods()) {
+            LOG.info(" {}",
+                     m);
+        }
+
+        File dir = new File("./target/JAVASSIST/");
+        dir.mkdirs();
+        // please note it is INTENTIONAL to write the file with package name part of the file itself, for easier browsing
+        // anyway the directory is NOT intended for classloading, but just for browsing bytecode for decompilation.
+        File bytecodeFile = new File(dir,
+                                     theCtClass.getPackageName() + theCtClass.getName() + ".class");
+        bytecodeFile.createNewFile();
+        FileOutputStream fos = new FileOutputStream(bytecodeFile);
+        fos.write(bytecode);
+        fos.close();
+        LOG.info("Written bytecode for {} in file: {}.",
+                 theCtClass.getName(),
+                 bytecodeFile);
+    }
+
+    @Test
+    public void testPojoWithCollectionsBytecode() {
+        PojoWithCollections pojo = new PojoWithCollections(new ArrayList(),
+                                                           new ArrayList(),
+                                                           new HashSet());
+
+        LOG.info("testPojoWithCollectionsBytecode(): {}",
+                 pojo);
+
+        assertEquals(pojo.getFieldCollection().getClass(),
```

Comment:
Looks like argument order of `assertEquals` is reversed. Should be `assertEquals(expected, actual)`